plugins/css-has-pseudo/src/browser.js

```diff
@@ -250,7 +241,7 @@ export default function cssHasPseudo(document, options) {
 						}
 
 						if (!options.mustPolyfill) {
-							rule.deleteRule();
+							styleSheet.deleteRule(index);
```

Comment:
This was the actual bug.
The fallback rules were never removed but the polyfill wasn't active because Chrome has native support.

Combined with `:not()` you can get funky results.

This changes correctly cleans up the fallback rules for browsers with native support.